src/sync-agent/sync-agent.ts

```diff
@@ -510,6 +512,7 @@ export class SyncAgent implements AsyncDisposable {
       latestOnly,
       onReset,
       groupKey,
+      snapshotTopic,
```

Comment:
You are passing an `undefined` to a default parameter. Though it works in JS/TS, I think this style may cause confusion. How about moving the default argument to `create()`.